### PR TITLE
SpinKitWave now have item count parameter

### PR DESCRIPTION
**Pull request by payam-zahedi**
Hi, I made some changes to SpinKitWave and make it more flexible. SpinKitWave now has `itemCount` property. It will help the user to choose number of waves. 

- `itemCount` property added to SpinKitWave Widget.
- `getAnimationDelay` now works dynamically. 
- and the `width` of each element is calculated according to `itemCount`.

check the demo in [DartPad](https://dartpad.dev/6f6dbe0fca5a77ad762cc1833e80df6f)

If you have any changes, please let me know.
thanks.

### CODE REVIEW

File: example/android/app/src/main/gen/com/example/example/BuildConfig.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*___Generated_by_IDEA___*/
```
Comment (jogboms) [inline example/android/app/src/main/gen/com/example/example/BuildConfig.java]: This file needs to be ignored

File: example/android/app/src/main/gen/com/example/example/Manifest.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*___Generated_by_IDEA___*/
```
Comment (jogboms) [inline example/android/app/src/main/gen/com/example/example/Manifest.java]: This file needs to be ignored

File: example/android/app/src/main/gen/com/example/example/R.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/*___Generated_by_IDEA___*/
```
Comment (jogboms) [inline example/android/app/src/main/gen/com/example/example/R.java]: This file needs to be ignored

File: lib/src/wave.dart [hunk 4]
```diff
@@ -10,15 +10,20 @@ class SpinKitWave extends StatefulWidget {
     this.type = SpinKitWaveType.start,
     this.size = 50.0,
     this.itemBuilder,
+    this.itemCount = 5,
     this.duration = const Duration(milliseconds: 1200),
     this.controller,
-  })  : assert(!(itemBuilder is IndexedWidgetBuilder && color is Color) && !(itemBuilder == null && color == null),
+  })  : assert(
```
Comment (jogboms) [inline lib/src/wave.dart]: We actually use lineWidth of 120 to improve readability

File: lib/src/wave.dart [hunk 5]
```diff
@@ -10,15 +10,20 @@ class SpinKitWave extends StatefulWidget {
...
+            !(itemBuilder is IndexedWidgetBuilder && color is Color) &&
+                !(itemBuilder == null && color == null),
             'You should specify either a itemBuilder or a color'),
+        assert(itemCount != null && itemCount >= 2, 'itemCount Cant be less then 2 '),
```
Comment (jogboms) [inline lib/src/wave.dart:18]: Can I make changes to the PR as well?

**Review by jogboms — COMMENTED** [review]
Good work @payam-zahedi. Thanks

File: lib/src/wave.dart (hunk 5 above)
Comment (payam-zahedi) [reply lib/src/wave.dart:18]: absolutely yes, should I give an access control?

Comment (jogboms) [reply lib/src/wave.dart:18]: I should be able to, just wanted to know if you are okay with. Thanks

Comment (payam-zahedi) [reply lib/src/wave.dart:18]: no problem, thanks for asking. good luck.

Comment (payam-zahedi) [reply lib/src/wave.dart:18]: hi @jogboms, is it possible to merge this PR?  I want to create new PR and I need new changes. 
if u need anything please let me know. thanks

Comment (jogboms) [reply lib/src/wave.dart:18]: Sorry i missed this comment. I would pick it up today. 🙏

**Review by jogboms — APPROVED** [review]
LGTM!

**Pull request merged**